src/Spice86/ViewModels/DisassemblyViewModel.Commands.cs

```diff
@@ -0,0 +1,183 @@
+namespace Spice86.ViewModels;
+
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.Mvvm.Messaging;
+
+using Spice86.Core.Emulator.VM.Breakpoint;
+using Spice86.Messages;
+using Spice86.Models.Debugging;
+using Spice86.Shared.Emulator.Memory;
+
+public partial class DisassemblyViewModel : IDisassemblyCommands {
+    // Explicitly implement IDisassemblyCommands interface
+    IAsyncRelayCommand IDisassemblyCommands.UpdateDisassemblyCommand => UpdateDisassemblyCommand;
+    IAsyncRelayCommand IDisassemblyCommands.NewDisassemblyViewCommand => NewDisassemblyViewCommand;
+    IRelayCommand IDisassemblyCommands.CopyLineCommand => CopyLineCommand;
+    IRelayCommand IDisassemblyCommands.StepIntoCommand => StepIntoCommand;
+    IRelayCommand IDisassemblyCommands.StepOverCommand => StepOverCommand;
+    IRelayCommand IDisassemblyCommands.GoToFunctionCommand => GoToFunctionCommand;
+    IRelayCommand IDisassemblyCommands.GoToCsIpCommand => GoToCsIpCommand;
+    IRelayCommand<SegmentedAddress?> IDisassemblyCommands.GoToAddressCommand => GoToAddressCommand;
+    IRelayCommand IDisassemblyCommands.CloseTabCommand => CloseTabCommand;
+    IRelayCommand<DebuggerLineViewModel> IDisassemblyCommands.CreateExecutionBreakpointHereCommand => CreateExecutionBreakpointHereCommand;
+    IRelayCommand<DebuggerLineViewModel> IDisassemblyCommands.RemoveExecutionBreakpointHereCommand => RemoveExecutionBreakpointHereCommand;
+    IRelayCommand<BreakpointViewModel> IDisassemblyCommands.DisableBreakpointCommand => DisableBreakpointCommand;
+    IRelayCommand<BreakpointViewModel> IDisassemblyCommands.EnableBreakpointCommand => EnableBreakpointCommand;
+    IRelayCommand<DebuggerLineViewModel> IDisassemblyCommands.ToggleBreakpointCommand => ToggleBreakpointCommand;
+    IRelayCommand IDisassemblyCommands.MoveCsIpHereCommand => MoveCsIpHereCommand;
+
+    [RelayCommand(CanExecute = nameof(CanCloseTab))]
+    private void CloseTab() {
+        _messenger.Send(new RemoveViewModelMessage<DisassemblyViewModel>(this));
+    }
+
+    [RelayCommand(CanExecute = nameof(IsPaused))]
+    private void StepOver() {
+        SegmentedAddress currentAddress = State.IpSegmentedAddress;
+        DebuggerLineViewModel debuggerLine = EnsureAddressIsLoaded(currentAddress);
+
+        // if (!debuggerLine.CanBeSteppedOver) {
```

Comment:
Dont forget to remove commented code

Reply:
It's actually restored now because this was work in progress.